Packet++/src/VrrpLayer.cpp

```diff
@@ -0,0 +1,533 @@
+		auto *vrrpPacketCommon = (vrrp_packet *) data;
+		uint8_t version = vrrpPacketCommon->version;
+		switch (version) {
+			case Vrrp_Version_2:
+				return new VrrpV2Layer(data, dataLen, prevLayer, packet);
+			case Vrrp_Version_3:
+				return new VrrpV3Layer(data, dataLen, prevLayer, packet, addressType);
+			default: {
+				return nullptr;
+			}
+		}
+	}
+
+	void VrrpLayer::computeCalculateFields() {
+		// calculate and fill the checksum to packet
+		fillChecksum();
+	}
+
+	uint8_t VrrpLayer::isIPv4Packet() const {
+		return (getAddressType() == IPAddress::IPv4AddressType);
+	}
+
+	uint8_t VrrpLayer::isIPv6Packet() const {
+		return (getAddressType() == IPAddress::IPv6AddressType);
+	}
+
+	uint8_t VrrpLayer::getIPAddressLen() const {
+		if (isIPv4Packet()) {
+			return 4;
+		} else if (isIPv6Packet()) {
+			return 16;
+		} else {
+			return 0;
+		}
+	}
+
+	uint8_t VrrpLayer::getVersionFromData() const {
+		vrrp_packet *packet = getPacketPtr();
+		return (packet->version);
+	}
+
+	void VrrpLayer::setPacket(vrrp_packet *packet) {
+		memcpy(m_Data, packet, sizeof(vrrp_packet));
+	}
+
+	bool VrrpLayer::isChecksumCorrect() const {
+		const vrrp_packet *packet = (vrrp_packet *) getData();
+		if (packet == nullptr) {
+			return false;
+		}
+
+		return (calculateChecksum() == be16toh(packet->checksum));
+	}
+
+	std::string VrrpLayer::getTypeName() const {
+		std::string typeName;
+		if (getPacketPtr()->type == VrrpType_Advertisement) {
+			typeName = "Advertisement";
+		} else {
+			typeName = "Unknown";
+		}
+
+		return typeName;
+	}
+
+	std::string VrrpLayer::getPriorityDesc() const {
+		std::string priorityName;
+		if (getPacketPtr()->priority == VRRP_PRIO_DFF) {
+			priorityName = "(Default priority for a backup VRRP router)";
+		} else if (getPriority() == VRRP_PRIO_STOP) {
+			priorityName = "(Current Master has stopped participating in VRRP)";
+		} else if (getPriority() == VRRP_PRIO_OWNER) {
+			priorityName = "(This VRRP router owns the virtual router's IP address(es))";
+		} else {
+		}
+
+		return priorityName;
+	}
+
+	std::string VrrpLayer::getAuthTypeDescByType(uint8_t authType) {
+		std::string priorityName;
+		if (authType == VRRP_AUTH_NONE) {
+			priorityName = "No Authentication";
+		} else if (authType == VRRP_AUTH_SIMPLE) {
+			priorityName = "Simple Text Authentication [RFC 2338] / Reserved [RFC 3768]";
+		} else if (authType == VRRP_AUTH_AH) {
+			priorityName = "IP Authentication Header [RFC 2338] / Reserved [RFC 3768]";
+		} else if (authType == VRRP_AUTH_MD5) {
+			priorityName = "Cisco VRRP MD5 authentication";
+		} else {
+		}
+
+		return priorityName;
+	}
+
+	/**
+		Virtual Router Redundancy Protocol
+		Version 2, Packet type 1 (Advertisement)
+		Virtual Rtr ID: 1
+		Priority: 100 (Default priority for a backup VRRP router)
+		Addr Count: 1
+		Auth Type: No Authentication (0)
+		Adver Int: 1
+		Checksum: 0xba52 [correct]
+		IP Address:
+			192.168.0.1
+	* */
+	std::string VrrpLayer::toString() const {
```

Comment:
`toString()` should consist of one line that describes the most important data of a layer, a little bit like Wireshark does:
![image](https://user-images.githubusercontent.com/9059541/231373857-dd4ea5ce-6c2c-4c14-9bff-286f1a01d2fb.png)

We should probably remove most of the data from here and keep just the version and maybe some more super important data, but definitely not all of it

Reply:
Please see this comment: https://github.com/seladb/PcapPlusPlus/pull/1118#discussion_r1165076754

Reply:
fixed here: https://github.com/seladb/PcapPlusPlus/pull/1118/commits/ddda563d5ded2fb531129b1ce85db68b578f39d1